File: bmi.rb
if ARGV.length != 2
  puts "Usage: bmi.rb <weight [kg]> <height [m]>"
  exit 1
end

inWeight, inHeight, *rest = ARGV

weight = inWeight.to_f
height = inHeight.to_f

bmi = weight / height ** 2

puts "Your BMI is " + bmi.to_s

if bmi <= 18.5
    puts "You're underweight."
elsif bmi <= 25
    puts "You're healthy."
elsif bmi <= 30
    puts "You're overweight."
else
    puts "Holy McShizzle, you're freaking whale!"
end

File: bmi.go
package main

import "os"
import "fmt"
import "strconv"

func main() {
	if len(os.Args) != 3 {
		fmt.Printf("Usage: %s <weight [kg]> <height [m]>\n", os.Args[0])
		os.Exit(0)
	}
	weight, _ := strconv.ParseFloat(os.Args[1], 64)
	height, _ := strconv.ParseFloat(os.Args[2], 64)
	bmi := weight / (height * height);

	fmt.Println("Your BMI is: ", bmi);

	if bmi <= 18.5 {
		fmt.Println("You're underweight.")
	} else if bmi <= 25.0 {
		fmt.Println("You're healthy.")
	} else if bmi <= 30.0 {
		fmt.Println("You're overweight.")
	} else {
		fmt.Println("Holy McSizzle, you're freaking whale!")
	}
}
